src/bindings/jna/plugins-kotlin/regex-example/src/README.md

```diff
@@ -0,0 +1,41 @@
+# Regex-Example Kotlin Plugin
+
+This plugin can be used to validate the values of keys via the `check/validation` metakey.
+It uses the value of the metakey as regex expression to check if newly added values match with the stored regex.
+If not, the plugin raises the `VALIDATION_SYNTACTIC` error code and prints the message from the `check/validation/message` metakey if present.
+
+#### Mounting the plugin
+
+To mount the plugin, use the following command:
+
+```sh
+kdb mount-java config.ni user:/test/process kdb:ni java:org.libelektra.plugin.RegexExamplePlugin
```

Comment:
This command did not work for me with the new Regex-plugin, but with the Java-plugins `Whitelist` and `sorted`.

I tried some things but, unfortunately, could not find a solution yet.

The error message is:
```
Was not able to load such a plugin!

Maybe you misspelled it, there is no such plugin or the loader has problems.
You might want to try to set LD_LIBRARY_PATH, use kdb-full or kdb-static.
Errors/Warnings during loading were:
Sorry, module process issued the error C01100:
Resource: Could not execute app (handshake read failed). Reason: Success
Mountpoint: /
Configfile: 
At: /home/flo/tu/tutor/P3/kotlin-binding/libelektra/src/plugins/process/process.c:153
 Sorry, 1 warning was issued ;(
 1: Module kdb issued the warning C01330:
	Plugin Misbehavior: Open of plugin returned unsuccessfully: process. Reason contains plugin, see other warnings for details
	Mountpoint: /
	Configfile: 
	At: /home/flo/tu/tutor/P3/kotlin-binding/libelektra/src/libs/elektra/plugin.c:101
```

Reply:
@kodebach do you have an idea?

Reply:
> `(handshake read failed). Reason: Success`

Well that seem odd...

Seems I didn't implement the `read` call for the handshake correctly:

https://github.com/ElektraInitiative/libelektra/blob/a7d16c82fabefd836595b70d7676a7c1b61a08a9/src/plugins/process/process.c#L149-L158

No idea why it only happens for this plugin and not the others. The handshake should always be the same.

I think something like this should work:

```c
	char handshakeAck[sizeof (MSG_HANDSHAKE_ACK_V1)];
        size_t totalSize = sizeof (MSG_HANDSHAKE_ACK_V1) - 1;
	ssize_t readBytes = 0;
        while (readBytes >= 0 && readBytes < totalSize) {
            readBytes += read (childToParent[0], &handshakeAck[readBytes], totalSize -  readBytes);
        }
	if (readBytes < 0)
	{
		ELEKTRA_SET_RESOURCE_ERRORF (errorKey, "Could not execute app (handshake read failed). Reason: %s", strerror (errno));
		close (parentToChild[1]);
		close (childToParent[0]);
		kill (pid, SIGTERM);
		return ELEKTRA_PLUGIN_STATUS_ERROR;
	}
```

Reply:
@Eiskasten did mounting work you?